packages/react-reconciler/src/ReactFiber.js

```diff
@@ -254,6 +254,23 @@ function FiberNode(
   this.alternate = null;
 
   if (enableProfilerTimer) {
+    // Note: The following is done to avoid a v8 deopt.
```

Comment:
Nit: `s/deopt/performance cliff/`

(It's not a deopt.)

Reply:
Interesting. I was referring to the fact that the result is Fibers all having different shapes (which seemed like a deopt to me) but admittedly, I was just guessing at terminology 😄

Reply:
To give some more background, a deopt happens when we're running optimized machine code generated by the optimizing compiler (TurboFan in V8's case), and one of the assumptions in the optimized code turns out to be invalid. In such cases we have to _deopt_ and go back to running bytecode in the interpreter (Ignition in V8's case). ([Our JSConf talk includes some more detail and visualizations that make this more clear, in case you're interested.](https://mathiasbynens.be/notes/shapes-ics))

This V8 bug however occurs before we even get to optimized code.

You can verify this yourself by running [our reduced test case](https://bugs.chromium.org/p/v8/issues/detail?id=8538) with `--no-opt`: it still reproduces.